### PR TITLE
Intmdb 628 datalakes profile changes

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

datalakes profile changes

Link to any related issue(s): 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [x] For CFN Resources: I have released by changes in the private registry and proved by change works in Atlas

### CODE REVIEW

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 1]
```diff
@@ -42,22 +42,24 @@ func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
 	}
 
-	// Create MongoDb Atlas Client using keys
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		return progressevents.GetFailedEventByCode(fmt.Sprintf("Error creating mongoDB client : %s", err.Error()),
-			cloudformation.HandlerErrorCodeInvalidRequest), nil
+	// Create atlas client
+	if currentModel.Profile == nil {
+		currentModel.Profile = aws.String(util.DefaultProfile)
```
Comment (andreaangiolillo) [inline cfn-resources/datalakes/cmd/resource/resource.go:47]:
```suggestion
		currentModel.Profile = aws.String(profile.DefaultProfile)
```

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 2]
```diff
@@ -94,13 +96,13 @@ func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
...
-	// Create MongoDb Atlas Client using keys
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		return handler.ProgressEvent{
-			OperationStatus:  handler.Failed,
-			Message:          err.Error(),
-			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	// Create atlas client
+	if currentModel.Profile == nil {
+		currentModel.Profile = aws.String(util.DefaultProfile)
```
Comment (andreaangiolillo) [inline cfn-resources/datalakes/cmd/resource/resource.go:101]:
```suggestion
		currentModel.Profile = aws.String(profile.DefaultProfile)
```

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 3]
```diff
@@ -136,14 +137,13 @@ func Update(req handler.Request, prevModel *Model, currentModel *Model) (handler
...
-	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
-	if err != nil {
-		_, _ = logger.Warnf("create - error: %+v", err)
-		return handler.ProgressEvent{
-			OperationStatus:  handler.Failed,
-			Message:          err.Error(),
-			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	// Create atlas client
+	if currentModel.Profile == nil {
+		currentModel.Profile = aws.String(util.DefaultProfile)
```
Comment (andreaangiolillo) [inline cfn-resources/datalakes/cmd/resource/resource.go:142]:
```suggestion
		currentModel.Profile = aws.String(profile.DefaultProfile)
```

File: cfn-resources/datalakes/cmd/resource/resource.go [hunk 4]
```diff
@@ -222,25 +222,24 @@ func List(req handler.Request, prevModel *Model, currentModel *Model) (handler.P
...
```
Comment (andreaangiolillo) [inline cfn-resources/datalakes/cmd/resource/resource.go:227]:
```suggestion
		currentModel.Profile = aws.String(profile.DefaultProfile)
```

File: cfn-resources/datalakes/mongodb-atlas-datalakes.json [hunk 5]
```diff
@@ -544,38 +544,34 @@
...
     "/properties/Hostnames",
     "/properties/State",
-    "/properties/EndDate",
-    "/properties/CloudProviderConfig/Aws/TestS3Bucket",
-    "/properties/NDSDataLakeDatabaseCollectionView",
-    "/properties/DataLakeDatabaseDataSourceView",
-    "/properties/DataLakeDatabaseView",
-    "/properties/NDSDataLakeDatabaseView",
-    "/properties/DataLakeAWSCloudProviderConfigView",
-    "/properties/DataLakeStorageView",
```
Comment (andreaangiolillo) [inline cfn-resources/datalakes/mongodb-atlas-datalakes.json:577]: why did you remove these props from the read only?

Comment (govardhanpagidi) [reply cfn-resources/datalakes/mongodb-atlas-datalakes.json:577]: @andreaangiolillo  We don't have all these values from server, except Aws/TestS3Bucket and this property is sent from Client.

Hence removed which are not coming from Server.

**Review by andreaangiolillo — APPROVED** [review]
LGTM
